Rubberduck.VBEEditor/SafeComWrappers/SafeIDispatchWrapper.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.SafeComWrappers
+{
+    public class SafeIDispatchWrapper<TDispatch> : SafeIDispatchWrapper
+    {
+        public SafeIDispatchWrapper(TDispatch target, bool rewrapping = false) : base(target, rewrapping)
+        { }
+
+        public new TDispatch Target => (TDispatch) base.Target;
+    }
+
+    public class SafeIDispatchWrapper : SafeComWrapper<dynamic>
+    {
+        public SafeIDispatchWrapper(object target, bool rewrapping = false) : base(target, rewrapping)
+        {
+            if (!Marshal.IsComObject(target))
+            {
+                throw new ArgumentException("The target object must be a COM object");
+            }
+
+            IDispatchPointer = GetPointer(target);
+        }
+
+        public IntPtr IDispatchPointer { get; }
+
+        public void Invoke(Action<dynamic> action)
+        {
+            action.Invoke(Target);
```

Comment:
Are we shielded against a dynamic binding exception here?

Reply:
Good question! I added a test in live code temporarily:

```
//testing
accessObject.Invoke(d => _logger.Trace("Dynamically invoked method:" + d.Name));
```

and got the output in the log:
```
2018-10-27 00:10:13.9527;TRACE-2.2.6874.274;Rubberduck.VBEditor.SafeComWrappers.SafeComWrapper`1;Dynamically invoked method:Form1;
```

So, no problem there with dispatching via `dynamic` in a lambda. Note that we provide the `dynamic`, rather than the caller. 

That said, the whole lambda is now dynamic and therefore, it'll have to be the caller's responsibility that there's appropriate handling if things don't go as expected.